pytorch_lightning/loops/optimizer/optimizer_loop.py

```diff
@@ -206,14 +214,16 @@ def zero_grad_fn():
         if not self._skip_backward and is_first_batch_to_accumulate:
             return zero_grad_fn
 
+        return None
```

Comment:
Can you reestructure this function?

```python

if _skip_backward:
    return


is_first_batch_to_accumulate = batch_idx % self.trainer.accumulate_grad_batches == 0
if is_first_batch_to_accumulate:
    def zero_grad_fn():
        self._on_before_zero_grad(optimizer)
        self._optimizer_zero_grad(batch_idx, optimizer, opt_idx)

    return zero_grad_fn
```

Reply:
done, but need the `return None` explicitly because mypy is strict